### PR TITLE
ngut37-sp2

### PR DESCRIPTION
**Pull request by ngut37**
https://sp-02.herokuapp.com/
- _automatic deployment from the [master branch of this repo](https://github.com/ngut37/sp02)_
- _keep in mind - FE and BE uses the free version of Heroku which means it has to start cold after a while of inactivity_

### How to run this app
- run `yarn` command in the root folder
- create a `.env` file in the root folder (variable names can be found in `.env-example`)
- if the [backend](https://github.com/ngut37/minibitly) is not cloned then use production API `REACT_APP_API_URL=https://minibitly.herokuapp.com`
- run `yarn start` (app listens on port `:3000`)

### Technical debt
- styling
- set domain to `shortify.name`
- about page
- feat: (spinning) loader
- feat: edit and delete links

### CODE REVIEW

File: www/ngut37/sp02/src/components/about/about.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import './about.scss';
+
+export const About = () => {
+  return (
+    <div className='aboutWrapper'>
```
Comment (nvbach91) [inline www/ngut37/sp02/src/components/about/about.tsx]: `"`

File: www/ngut37/sp02/src/app.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import { Navbar } from './components/navbar/navbar';
+import { AddForm } from './components/add-form/form';
+import { LinksList } from './components/links-list/links-list.';
+import { About } from './components/about/about';
+import { Footer } from './components/footer/footer';
+
+export const App = () => {
+  return (
+    <Router>
```
Comment (nvbach91) [inline www/ngut37/sp02/src/app.tsx:13]: proc se vase aplikace refreshuje kdyz kliknu na navigaci?

File: www/ngut37/sp02/src/components/navbar/navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import './navbar.scss';
+
+export const Navbar = () => {
+  return (
+    <div className="headerWrapper">
+      <span className="logo"><a href="./">Shortify</a></span>
+      <nav>
+        <a href="./">Create link</a>
+        <a href="./list">Find your path</a>
```
Comment (nvbach91) [inline www/ngut37/sp02/src/components/navbar/navbar.tsx]: aha

File: www/ngut37/sp02/src/components/footer/footer.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+import './footer.scss';
+
+export const Footer = () => {
+  return (
+    <div className="footerWrapper">
+      <span>React App made with ❤</span>
+      <span><a href="https://github.com/ngut37">@ngut37</a></span>
```
Comment (nvbach91) [inline www/ngut37/sp02/src/components/footer/footer.tsx]: `target="?"`

**Review by nvbach91 — CHANGES_REQUESTED** [review]
- nevim jak tuto aplikaci pouzit
- sjednotte ty 3 page do single page
- zmensit obrazek...
- dalsi postup az na zkousce...

File: www/ngut37/sp02/src/components/about/about.tsx (hunk 1 above)
Comment (ngut37) [reply www/ngut37/sp02/src/components/about/about.tsx]: Opraveno

File: www/ngut37/sp02/src/app.tsx (hunk 2 above)
Comment (ngut37) [reply www/ngut37/sp02/src/app.tsx:13]: Opraveno pomoci komponenty `<Link>`

File: www/ngut37/sp02/src/components/navbar/navbar.tsx (hunk 3 above)
Comment (ngut37) [reply www/ngut37/sp02/src/components/navbar/navbar.tsx]: Viz komentar nahore

File: www/ngut37/sp02/src/components/footer/footer.tsx (hunk 4 above)
Comment (ngut37) [reply www/ngut37/sp02/src/components/footer/footer.tsx]: Opraveno

**Comment by ngut37** [comment]
Redeploy na https://sp-02.herokuapp.com/

- [x] URL preview - implementovan, misto `onChange` jsem to dal `onBlur`, aby to nebombardovalo backend pri kazde zmene (deploynul jsem novou featuru z backendoveho repozitare)
- [x] localStorage - zaznam se uklada do `localStorage` kdyz uzivatel clickne na **Create Link**, pokud pod stejny namepath napise jine heslo, tak se tomu zaznamu prepise (tzn. nevytvari se novy zaznam se stejnym namespace)
- [x] drobne stylove upravy a odstraneni obrazku

**Review by nvbach91 — APPROVED** [review]
how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638

**Pull request merged**